api/questions/ridetrips.py

```diff
@@ -118,4 +118,35 @@ def get_total_trips_by_dropoff_specific_area_and_year(self, year, pickup_area):
         cur = self.con.cursor()
         cur.execute(query)
         rows = rows_to_dicts(cur, cur.fetchall())
+        return rows
+        
+    def community_areas(self):
+        """
+        Returns all of the community areas.
+        """
+        query = """
+        SELECT
+            area_number,
+            name,
+            part
+        FROM community_area
+        """
+        cur = self.con.cursor()
+        cur.execute(query)
+        rows = rows_to_dicts(cur, cur.fetchall())
+        return rows
+
+    def metrics_by_area(self,rows):
+        """
+        Returns the rows, but with a new column that contains the area names
+        """
+        ls = self.community_areas()
+        
+        for area in ls:
+            y = area['area_number'] 
+            #x = rows[y-1]['dropoff_community_area']
+            x = rows[y-1]['dropoff_community_area']
+            if x == y:
+                rows[y-1]['area_name'] = area['name']
```

Comment:
If x is not equal to y (in the case that `rows` is not sorted, you will miss some area names.

Two options:
1. Revise this Python method
2. Add area name in the SQL by doing a left join

I think option 1 would work fine. If you do option 2, then you would also need to add some fake community areas to your unit test, which might be extra work for little benefit.

Here's the pseudocode I would recommend for option 1:
- Convert the list of community areas into a dictionary where the key is the area number and the value is the object with the name, number, and part.
- Loop over all the values in rows and look up the dropoff area in the dictionary
- Add the name, number, and part info to the output (or just the name if that's all you need)

Reply:
With option 1:
Will this suffice?
`    def metrics_by_area(self,rows):
        """
        Returns the rows, but with a new column that contains the area names
        """
        res = {}
        
        for area in self.community_areas():
            res[area["area_number"]] = area
        for row in rows:
            if row['dropoff_community_area'] in res.keys():
                row['area_name'] = res.get(row['dropoff_community_area'])['name']
                row['part'] = res.get(row['dropoff_community_area'])['part']
        return rows`

I have tested this and it still works just fine for the frontend